lib/taskana-core-test/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java

```diff
@@ -3248,6 +3261,283 @@ void should_ReturnEmptyList_When_QueryingForNonexistentSystemInstanceLike() {
+            };
+
+        return DynamicTest.stream(testCases.iterator(), Triplet::getLeft, test);
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @TestFactory
+      Stream<DynamicTest> should_ApplyFilter_When_QueryingForCustomIntWithin() {
+        List<Quadruple<String, TaskCustomIntField, IntInterval, IntInterval>> testCases =
+            List.of(
+                Quadruple.of(
+                    "CustomInt1",
+                    TaskCustomIntField.CUSTOM_INT_1,
+                    new IntInterval(1, 1),
+                    new IntInterval(-10, -8)),
+                Quadruple.of(
+                    "CustomInt2",
+                    TaskCustomIntField.CUSTOM_INT_2,
+                    new IntInterval(0, null),
+                    new IntInterval(0, null)),
+                Quadruple.of(
+                    "CustomInt3",
+                    TaskCustomIntField.CUSTOM_INT_3,
+                    new IntInterval(-1, 5),
+                    new IntInterval(0, null)),
+                Quadruple.of(
+                    "CustomInt4",
+                    TaskCustomIntField.CUSTOM_INT_4,
+                    new IntInterval(0, 1),
+                    new IntInterval(-2, 22)),
+                Quadruple.of(
+                    "CustomInt5",
+                    TaskCustomIntField.CUSTOM_INT_5,
+                    new IntInterval(3, 9),
+                    new IntInterval(-1000, -50)),
+                Quadruple.of(
+                    "CustomInt6",
+                    TaskCustomIntField.CUSTOM_INT_6,
+                    new IntInterval(4, 9),
+                    new IntInterval(0, null)),
+                Quadruple.of(
+                    "CustomInt7",
+                    TaskCustomIntField.CUSTOM_INT_7,
+                    new IntInterval(4, 9),
+                    new IntInterval(11, 22)),
+                Quadruple.of(
+                    "CustomInt8",
+                    TaskCustomIntField.CUSTOM_INT_8,
+                    new IntInterval(123, 124),
+                    new IntInterval(1, null)));
+
+        ThrowingConsumer<Quadruple<String, TaskCustomIntField, IntInterval, IntInterval>> test =
+            q -> {
+              List<TaskSummary> result =
+                  taskService
+                      .createTaskQuery()
+                      .workbasketIdIn(wb.getId())
+                      .customIntAttributeWithin(q.getSecond(), q.getThird(), q.getFourth())
+                      .list();
+              assertThat(result).containsExactly(taskSummary1);
+            };
+
+        return DynamicTest.stream(testCases.iterator(), Quadruple::getFirst, test);
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @TestFactory
+      Stream<DynamicTest> should_ApplyFilter_When_QueryingForCustomIntNotWithin() {
+        List<Quadruple<String, TaskCustomIntField, IntInterval, IntInterval>> testCases =
+            List.of(
+                Quadruple.of(
+                    "CustomInt1",
```

Comment:
* replace string with enum for descrition
* use IntInterval[]

Reply:
Ok :)